src/main/java/br/net/silva/daniel/payment/challenge/simplify/payment/transfer/api/infrastructure/controller/TransferController.java

```diff
@@ -0,0 +1,27 @@
+package br.net.silva.daniel.payment.challenge.simplify.payment.transfer.api.infrastructure.controller;
+
+import br.net.silva.daniel.payment.challenge.simplify.payment.transfer.api.infrastructure.controller.request.TransferRequestTransaction;
+import br.net.silva.daniel.payment.challenge.simplify.payment.transfer.api.infrastructure.controller.response.TransferResponse;
+import br.net.silva.daniel.payment.challenge.simplify.payment.transfer.api.infrastructure.service.TransferService;
+import jakarta.validation.Valid;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/transfer")
+public class TransferController {
+
+    private final TransferService service;
+
+    public TransferController(TransferService service) {
+        this.service = service;
+    }
+
+    @PostMapping
+    public ResponseEntity<TransferResponse> transfer(@RequestBody @Valid TransferRequestTransaction request) throws Exception {
+        return ResponseEntity.ok(service.transfer(request));
```

Comment:
**Suggestion:** Modify the `transfer` method to catch and handle specific exceptions such as `AccountNotFoundException` and `IllegalArgumentException`, providing more detailed HTTP responses based on the exception type. [enhancement, importance: 7]
```suggestion
    public ResponseEntity<TransferResponse> transfer(@RequestBody @Valid TransferRequestTransaction request) {
        try {
            return ResponseEntity.ok(service.transfer(request));
        } catch (AccountNotFoundException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(new TransferResponse(ex.getMessage()));
        } catch (IllegalArgumentException ex) {
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new TransferResponse(ex.getMessage()));
        } catch (Exception ex) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(new TransferResponse("An error occurred"));
        }
    }
```